graphql_compiler/compiler/helpers.py

```diff
@@ -510,6 +510,24 @@ def get_first_folded_edge(self):
         first_folded_edge_direction, first_folded_edge_name = self.fold_path[0]
         return first_folded_edge_direction, first_folded_edge_name
 
+    def get_full_path_location_name(self):
```

Comment:
Would it harm anything if this approach (or a similar one) were made to be the standard `get_location_name()` implementation for this type, instead of having a separate method for it? The comment on lines 489-495 may have been true when written, but it doesn't seem to be true anymore given the need for this new function, so it may be time to revisit.

Reply:
This is specifically useful for when we need to traverse while within a fold scope, i.e. the fold directive isn't at the innermost scope. For gremlin and match we only need to name the vertex at the end of the path, so it's easy to get away with using a different naming system-- but if we replace `get_location_name()` with this implementation and modify the compiler tests, it shouldn't be too bad (since it's just a naming change). 

Could be a worthwhile thing to do after fold is implemented, to get consistent naming for the different backends in test_compiler.py